screen2.0/src/app/_scatterPlot/scatterPlot.tsx

```diff
@@ -10,23 +10,35 @@ import { useDrag } from '@visx/drag';
 import CircularProgress from '@mui/material/CircularProgress';
 import { curveBasis } from '@visx/curve';
 import { Zoom } from '@visx/zoom';
+import { createPortal } from 'react-dom';
 
+/*
+    All information given to a point on the plot, including its coordinates(x and y), its radius, color, and opacity, and its metadata information
+    which can be any amount of strings used to display in the tooltip
+*/
 interface Point {
     x: number;
     y: number;
     r?: number;
     color: string;
     opacity?: number;
-    name: string;
-    accession: string;
+    metaData: Record<string, any>;
```

Comment:
IMO I think the better approach would be to use generic types for this. Using Record allows you to pass objects of whatever form (and then you're iterating through them for the tooltip), but generics and type arguments would allow you do the same but then have access to the correct type of whatever you pass for as metaData in your `onSelectionChange` and a hypothetical "`onPointClicked`" function. That is unless I'm missing something, not super familiar with Record. I think with Record, you'll always get `any` as the type since it's `Record<string, any>`

Reply:
Also if this is to be a generic scatterplot, I think a function something like `tooltipBody: (point: Point[]) => JSX.Element` would be very helpful to allow control over what does and doesn't appear in the tooltip, and to format data. For example, if a point has a p-val of 0.0000000035, I want to be able to format that as 3.5 x 10^-9. Or if I am displaying a gene name, I want to italicize it.

Having a fallback that iterates though metadata is good though